### ISSUE TITLE
http_server on port 9001 closing down after a while. 

### ISSUE THREAD

**Issue by mindfuucker** [Supervisor/supervisor#1196]
Hello Github,

I am using version 4.0.0.dev0 on a headless pi 3B+ running raspbian lite.
Supervisord is used to reverse tunnel some local ports (including the http_server) to the outside world:
```
[unix_http_server]
file=/tmp/supervisor.sock   ; the path to the socket file
;chmod=0700                 ; socket file mode (default 0700)
;chown=nobody:nogroup       ; socket file uid:gid owner
;username=user              ; default is no username (open server)
;password=123               ; default is no password (open server)

[inet_http_server]         ; inet (TCP) server disabled by default
port=*:9001        ; ip_address:port specifier, *:port for all iface
username=user              ; default is no username (open server)
password=pass               ; default is no password (open server)

[supervisord]
logfile=/tmp/supervisord.log ; main log file; default $CWD/supervisord.log
logfile_maxbytes=50MB        ; max main logfile bytes b4 rotation; default 50MB
logfile_backups=0            ; # of main logfile backups; 0 means none, default 10
loglevel=debug                ; log level; default info; others: debug,warn,trace
pidfile=/tmp/supervisord.pid ; supervisord pidfile; default supervisord.pid
nodaemon=false               ; start in foreground if true; default false
minfds=1024                  ; min. avail startup file descriptors; default 1024
minprocs=200                 ; min. avail process descriptors;default 200
environment=AUTOSSH_MAXSTART="4",AUTOSSH_DEBUG="0",AUTOSSH_POLL="60"

[supervisorctl]
;serverurl=unix:///tmp/supervisor.sock ; use a unix:// URL  for a unix socket
serverurl=http://127.0.0.1:9001 ; use an http:// url to specify an inet socket
username=user              ; should be same as in [*_http_server] if set
password=pass                ; should be same as in [*_http_server] if set
;prompt=mysupervisor         ; cmd line prompt (default "supervisor")
;history_file=~/.sc_history  ; use readline history if available

[program:remoteAccess_2002]
command=/usr/bin/ssh -N -R 2002:localhost:22000 pi@some_ip_address
autostart=false
autorestart=true
startretries=1

[program:thisInterface_2000]
command=/usr/bin/ssh -N -R 2000:localhost:9001 pi@some_ip_address
autostart=true
autorestart=true
startretries=1
```
All functions fine for a while, but after a couple of days the http server just drops out. 
Testing the server gives me:
```
pi@external:~ $ nc -vz 127.0.0.1 9001
nc: connect to 127.0.0.1 port 9001 (tcp) failed: Connection refused
```
Checking the log at /tmp/supervisord.log yields an empty file. Checking the program error log gives me:
```
sudo nano /tmp/thisInterface_2000-stderr---supervisor-ungpi49q.log

connect_to localhost port 9001: failed.
connect_to localhost port 9001: failed.
connect_to localhost port 9001: failed.
connect_to localhost port 9001: failed.
```
Status gives me:
```
pi@external:~ $ supervisorctl status
http://127.0.0.1:9001 refused connection
```
Reload gives: 
```
supervisorctl reload
error: <class 'ConnectionRefusedError'>, [Errno 111] Connection refused: file: /usr/local/lib/python3.6/socket.py line: 713
```
This indicates supervisord was still running but the server was down. The SSH tunnel is also still running so its just the HTTP server that is down.

**Comment by mnaberez** [comment]
> I am using version 4.0.0.dev0
...
> All functions fine for a while, but after a couple of days the http server just drops out.

What git rev are you using (`git rev-parse HEAD` or the top of `git log`) and what Python version?  A very similar issue was reported in #1178 and #1157 that was fixed just recently.  If you are running a rev before 2a5fd123ff96630f1e9584083f22da7287706110 then it likely has already been fixed.

**Comment by mindfuucker** [comment]
Hi,

I am using Python 3.6.5. The rev is difficult to determine because I used 
`pip install git+https://github.com/Supervisor/supervisor`

My GIT skills are horrible and I do not know where to find anything. I installed it a couple of months ago, so it might have been fixed already. I will upgrade with:
`pip install --upgrade git+https://github.com/Supervisor/supervisor`

**Comment by mnaberez** [comment]
> I am using Python 3.6.5.
...
> I installed it a couple of months ago 

This issue sounds just like #1178 and #1157, which was fixed in 2a5fd123ff96630f1e9584083f22da7287706110 just over one month ago.  A couple of months ago would have been a rev before that fix was merged.  I'm going to close this issue because of that.  Please upgrade to a newer rev.  If you still have the issue after upgrading, please comment here and we can reopen.
